general/device.cpp

```diff
@@ -24,21 +24,21 @@ namespace mfem
 namespace internal
 {
 
-OccaDevice occaDevice;
-
 // Backends listed by priority, high to low:
 static const Backend::Id backend_list[Backend::NUM_BACKENDS] =
 {
-   Backend::OCCA_CUDA, Backend::RAJA_CUDA, Backend::CUDA,
+   Backend::OCCA_CUDA, Backend::RAJA_CUDA, Backend::CUDA, Backend::CUDA_UVM,
```

Comment:
It's good to see a way to support UVM coming in.  If worse comes to worst we could go for HYPRE GPU support with this.

Reply:
It's not in any of these changes, but I took a look at the code for the Push and Pull when using UVM.  I noticed that you are calling cudaMemcpy for those which could cause trouble when one of the memory spaces is oversubscribed.  I would recommend switching these to cudaMemAdvise.